homeassistant/components/trace/__init__.py

```diff
@@ -0,0 +1,12 @@
+"""Support for automation and script tracing and debugging."""
+from . import websocket_api
+from .const import DATA_TRACE
+
+DOMAIN = "trace"
+
+
+async def async_setup(hass, config):
+    """Initialize the trace integration."""
+    hass.data.setdefault(DATA_TRACE, {})
```

Comment:
Don't use `setdefault`. We only call `async_setup` guaranteed once.
```suggestion
    hass.data[DATA_TRACE] = {}
```

Reply:
Fixed in #48276